### PR TITLE
feature/max_targets

### PR DESCRIPTION
**Pull request by Rodrigo-Miranda-0**
#### Board:
* [Ticket #g4IaYJN1](https://trello.com/c/g4IaYJN1/5-2-as-a-user-i-should-only-be-able-to-create-a-maximum-of-10-targets-so-the-map-doesnt-get-too-crowded)
---
#### Description:

* Limit the amount of targets a user can create to 10
---
#### Notes:
*
---
#### Tasks:
- [x] Create validation of max targets
---
#### Risk:
*
---
#### Preview:

### CODE REVIEW

File: app/services/api/v1/target_service.rb [hunk 1]
```diff
@@ -9,6 +9,8 @@ def initialize(target_params, current_user, longitude, latitude)
       end
 
       def create
+        raise MaxTargetError if @current_user.targets.count >= Target::MAX_TARGETS
```
Comment (enzofab91) [inline app/services/api/v1/target_service.rb:12]: What about adding this validation in the model instead of the controller? So that if later you needs to create a target from a service or something like that you don't have to validate it again there. wdyt?

Comment (jpascual1994) [reply app/services/api/v1/target_service.rb:12]: I think it's ok, we should only create targets by using the service object. If we need to create targets from another service, we can still use this service.

File: spec/requests/api/v1/targets/create_spec.rb [hunk 2]
```diff
@@ -74,5 +75,25 @@
...
+        expect(response).to have_http_status(:unprocessable_entity)
+      end
+
+      it 'should not add the target to the database' do
+        expect { subject }.not_to change(Target, :count)
+      end
+
+      it 'should return the error message (Max targets reached)' do
+        subject
+        p response.body
```
Comment (jpascual1994) [inline spec/requests/api/v1/targets/create_spec.rb]: Please remove this line

File: spec/requests/api/v1/targets/create_spec.rb [hunk 3]
```diff
@@ -74,5 +75,25 @@
         )
       end
     end
+
+    context 'with max targets reached' do
+      let(:headers) { user2.create_new_auth_token }
```
Comment (jpascual1994) [inline spec/requests/api/v1/targets/create_spec.rb:113]: Please add an empty line between the last let and the first it

File: spec/requests/api/v1/targets/create_spec.rb [hunk 4]
```diff
@@ -74,5 +75,25 @@
         )
       end
     end
+
+    context 'with max targets reached' do
```
Comment (juanmferreira93) [inline spec/requests/api/v1/targets/create_spec.rb:112]: I probably create different `context` to test edge values.. like
- when I have 9 Targets and try to create the number 10 (it should allow me)
- when I have 10 Target and try to create the number 11 (it should raise an error) --> this is the one you implemented I think

**Review by juanmferreira93 — APPROVED** [review]

**Review by jpascual1994 — APPROVED** [review]
